lib/Rest/GeneralRequest.h

```diff
@@ -84,6 +84,8 @@ class GeneralRequest {
         _requestContext(nullptr),
         _isRequestContextOwner(false),
         _authenticated(false),
+        _authActive(true), // the default is true as in some cases
+                           // not active implies authenticated
```

Comment:
Why is it necessary to add a flag here? Authentication is either on or off for the entire server ?!

Reply:
Why should we include the server feature allover the place? Is there a reason why we do not simply set authenticated to true authentication is disabled?

Reply:
yes. Please do not add pointless flags to requests. Use "ExecContext::isAuthEnabled()" or simply  make it a convenience method checking `AuthenticationFeature`

Reply:
That does not answer at all why requests can be unauthorized when the feature responsible for authorization is disabled. Passing the bool should not be necessary at all - I totally agree. But what is good about the monolithic design that constantly uses global state and requires each part of the code to know about the other.